wavenet.py

```diff
@@ -53,16 +56,26 @@ def _create_dilation_layer(self, input_batch, layer_index, dilation, in_channels
 
         out = tf.tanh(conv_filter) * tf.sigmoid(conv_gate)
 
-        weights_dense = tf.Variable(tf.truncated_normal(
-            [1, dilation_channels, in_channels], stddev=0.2, name="dense"))
-        transformed = tf.nn.conv1d(out, weights_dense, stride=1,
-                                   padding="SAME", name="dense")
-        layer = 'layer{}'.format(layer_index)
-        tf.histogram_summary(layer + '_filter', weights_filter)
-        tf.histogram_summary(layer + '_gate', weights_gate)
-        tf.histogram_summary(layer + '_dense', weights_dense)
+        # The result of this 1x1 convolution is passed to the next block.
+        weights_residuals = tf.Variable(tf.truncated_normal(
+            [1, dilation_channels, in_channels], stddev=0.2, name="residuals"))
+        residuals = tf.nn.conv1d(out, weights_residuals, stride=1,
+                                 padding="SAME", name="residuals")
 
-        return transformed, input_batch + transformed
+        # The result of this 1x1 convolution is gathered at the end.
+        weights_skipped = tf.Variable(tf.truncated_normal(
+            [1, dilation_channels, in_channels], stddev=0.2, name="skipped"))
```

Comment:
I suggest adding a separate skip_channels param that serves as the number of output channels, instead of "in_channels" AKA residual_channels. I think we want the flexibility to have the skip contribution channels be different than the residual channels.